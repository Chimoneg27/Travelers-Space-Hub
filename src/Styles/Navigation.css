@import url('https://fonts.googleapis.com/css2?family=Inter:wght@400;600;700&family=Open+Sans&family=Roboto&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

:root {
  --azure: #0077b6;
  --powder-blue: #2ea8d1;
  --sky-blue: #206fdf;
}

.navigation {
  width: 95%;
  display: flex;
  justify-content: space-between;
  padding: 1.5rem;
  margin: 0 auto;
  font-family: 'Inter', sans-serif;
}

.Logo {
  display: flex;
  align-items: center;
  gap: 1rem;
}

.Logo img {
  width: 4rem;
}

.items {
  display: flex;
  justify-content: space-between;
  align-items: center;
  list-style-type: none;
  width: 24%;
  padding: 0.15rem;
  color: var(--azure);
}

.items li {
  font-size: 1.1rem;
  font-weight: 600;
  cursor: pointer;
}

.items li:last-child {
  border-left: 2px solid black;
  padding-left: 1rem;
}

.noactive-link {
  color: var(--powder-blue);
  text-decoration: none;
}

.active {
  color: var(--sky-blue);
  border-bottom-width: 1px;
  border-bottom-style: solid;
}
