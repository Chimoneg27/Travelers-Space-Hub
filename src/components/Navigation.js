import React from 'react';
import { NavLink } from 'react-router-dom';
import '../Styles/Navigation.css';
import logoImage from '../assets/Space-logo.jpg';

const Navigation = () => (
  <>
    <nav className="navigation">
      <div className="Logo">
        <img src={logoImage} alt="logo" />
        <h2>Space Travelers&apos; Hub</h2>
      </div>
      <ul className="items">
        <li><NavLink to="/" activeclassname="active-link" className="noactive-link">Rockets</NavLink></li>
        <li><NavLink to="/Missions" activeclassname="active-link" className="noactive-link">Missions</NavLink></li>
        <li><NavLink to="/My Profile" activeclassname="active-link" className="noactive-link">My Profile</NavLink></li>
      </ul>
    </nav>
    <div className="line" />
  </>
);

export default Navigation;
