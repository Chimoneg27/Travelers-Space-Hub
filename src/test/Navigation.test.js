import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import Navigation from '../components/Navigation';

const links = [
  {
    id: 1,
    path: '/',
    nav: 'Rockets',
  },
  {
    id: 2,
    path: '/missions',
    nav: 'Missions',
  },
  {
    id: 3,
    path: '/myProfile',
    nav: 'My Profile',
  },
];

const App = () => (
  <BrowserRouter>
    <Navigation links={links} />
  </BrowserRouter>
);

describe('Test Navigation', () => {
  it('should Navigation all links', () => {
    expect(screen.getByText('')).toMatchSnapshot();
    render(<App />);
    expect(Navigation).toMatchSnapshot();
  });
});
