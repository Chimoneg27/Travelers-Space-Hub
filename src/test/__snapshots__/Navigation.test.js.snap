// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navigation should Navigation all links 1`] = `<body />`;

exports[`Test Navigation should Navigation all links 2`] = `[Function]`;
